PostHog/PostHogSessionManager.swift

```diff
@@ -12,8 +12,11 @@ class PostHogSessionManager {
 
     private let anonLock = NSLock()
     private let distinctLock = NSLock()
+    private let idGen: (UUID) -> UUID
+
     init(_ config: PostHogConfig) {
         storage = PostHogStorage(config)
+        idGen = config.getAnonymousId
```

Comment:
```suggestion

    init(_ config: PostHogConfig) {
        storage = PostHogStorage(config)
```

Reply:
@marandaneto was not able to do this because `config` was not saved as a field so the private method below cannot refer to it.